src/Components/ModuleForm/ModuleForm.js

```diff
@@ -0,0 +1,310 @@
+import React, { Component } from "react";
+import "./ModuleForm.css";
+import { Button, ListGroup, Card } from "react-bootstrap";
+
+class ModuleForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      enabled: "1",
+      userAction: "Add",
+      moduleName: "",
+      moduleList: JSON.parse(localStorage.getItem("moduleList")) || [],
+      grade: "A",
+      error: "",
+    };
+  }
+
+  handleSubmit = (event) => {
+    const { userAction, moduleList, moduleName, grade } = this.state;
+
+    console.log("handleSubmit");
+
+    if (moduleName === "") {
+      this.setState({
+        error: "Module Name cannot be empty",
+      });
+      event.preventDefault();
+      return;
+    }
+    // Add logic for adding modules
+    if (userAction === "Add") {
```

Comment:
I think since you have a couple of user actions, you could just create functions for those. 
`addModule()`, `editModule()` etc.

Also, since you are checking against a static string, I would recommend creating ENUM strings for those, so no typos will ever sneak into your code. Example:

```
const USER_ACTIONS = {
    ADD: "Add",
    UPDATE: "Update"
}
```

You could even go further and describe the values as the functions:

```
const USER_ACTIONS = {
      ["Add"]: addModule(),
      ["Update"]: updateModule()
}
```

Then you don't need to run if/else spaghetti but can just do USER_ACTIONS[userAction] and it will call the relevant function